<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Anthony Hung" />

<meta name="date" content="2021-01-19" />

<title>Integration and clustering of scRNA sequencing data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">invitroOA_pilot_repository</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Integration and clustering of scRNA sequencing data</h1>
<h4 class="author">Anthony Hung</h4>
<h4 class="date">2021-01-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>invitroOA_pilot_repository/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree0ef10b172cb29c1b75705d1f6dfbe2e8fcc74fbctargetblank0ef10b1a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/0ef10b172cb29c1b75705d1f6dfbe2e8fcc74fbc" target="_blank">0ef10b1</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree0ef10b172cb29c1b75705d1f6dfbe2e8fcc74fbctargetblank0ef10b1a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/0ef10b172cb29c1b75705d1f6dfbe2e8fcc74fbc" target="_blank">0ef10b1</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/conda-archive/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/conda/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/locks/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/shadow/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/singularity/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/tmp.3ekfs3n5/
    Ignored:    code/bulkRNA_preprocessing/fastq/
    Ignored:    code/bulkRNA_preprocessing/out/
    Ignored:    code/single_cell_preprocessing/.snakemake/conda-archive/
    Ignored:    code/single_cell_preprocessing/.snakemake/conda/
    Ignored:    code/single_cell_preprocessing/.snakemake/locks/
    Ignored:    code/single_cell_preprocessing/.snakemake/shadow/
    Ignored:    code/single_cell_preprocessing/.snakemake/singularity/
    Ignored:    code/single_cell_preprocessing/YG-AH-2S-ANT-1_S1_L008/
    Ignored:    code/single_cell_preprocessing/YG-AH-2S-ANT-2_S2_L008/
    Ignored:    code/single_cell_preprocessing/demuxlet/.DS_Store
    Ignored:    code/single_cell_preprocessing/fastq/
    Ignored:    data/external_scRNA/Chou_et_al2020/
    Ignored:    data/external_scRNA/Jietal2018/
    Ignored:    data/external_scRNA/Wuetal2021/
    Ignored:    data/external_scRNA/merged_external_scRNA.rds
    Ignored:    data/poweranalysis/alasoo_et_al/
    Ignored:    output/GO_terms_enriched.csv
    Ignored:    output/topicModel_k=7.rds
    Ignored:    output/voom_results.rds

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/clustering_scRNA.Rmd</code>) and HTML (<code>docs/clustering_scRNA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0ef10b172cb29c1b75705d1f6dfbe2e8fcc74fbc/analysis/clustering_scRNA.Rmd" target="_blank">0ef10b1</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
warning = F
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/0ef10b172cb29c1b75705d1f6dfbe2e8fcc74fbc/docs/clustering_scRNA.html" target="_blank">0ef10b1</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
warning = F
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/analysis/clustering_scRNA.Rmd" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
suppress warnings
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/clustering_scRNA.html" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
suppress warnings
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/d7cac0172263e10ce3a5193d0fc91ffafba8ffbb/analysis/clustering_scRNA.Rmd" target="_blank">d7cac01</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
load more libraries
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/5e6c873d8560b1a2a8498bc7e1a5c5304da815f3/analysis/clustering_scRNA.Rmd" target="_blank">5e6c873</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
load ggplot2
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/37a702d29509d17181eaa450b1c271494ac244ad/analysis/clustering_scRNA.Rmd" target="_blank">37a702d</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
add clustering details
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/78cfbcd3204249a8673228d684ae77444c63266b/analysis/clustering_scRNA.Rmd" target="_blank">78cfbcd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
finish paring down files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/99c70b8e43338b294c4aa7877ea7eccc732be2ac/analysis/clustering_scRNA.Rmd" target="_blank">99c70b8</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
update external data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/d6b5b17900d3314edb102b088ef4a7b12cf2a6ca/analysis/clustering_scRNA.Rmd" target="_blank">d6b5b17</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
Added analysis files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/28f57faed7255644b4858290aed125d5466312ea/analysis/clustering_scRNA.Rmd" target="_blank">28f57fa</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-19
</td>
<td>
Add files for analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>To determine cell type heterogeneity between samples jointly, this code treats each individual (unstrained condition) as its own separate sample and performs integration across individuals. It then finds clusters within the integrated data and characterizes the clusters.</p>
</div>
<div id="load-data-and-packages" class="section level1">
<h1>Load data and packages</h1>
<p>The ANT1.2 seurat object was generated by running the preprocessing code in <a href="preProcess_scRNA.html">Pre-processing of raw 10x files into count matrices and demultiplexing</a>.</p>
<pre class="r"><code>library(Seurat)
library(gridExtra)
library(ggplot2)
library(cowplot)
library(tidyverse)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;cli&#39;:
  method     from    
  print.boxx spatstat</code></pre>
<pre><code>── Attaching packages ────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ tibble  3.0.4     ✓ dplyr   1.0.2
✓ tidyr   1.1.2     ✓ stringr 1.4.0
✓ readr   1.3.1     ✓ forcats 0.4.0
✓ purrr   0.3.4     </code></pre>
<pre><code>── Conflicts ───────────────────────────────────── tidyverse_conflicts() ──
x dplyr::combine() masks gridExtra::combine()
x dplyr::filter()  masks stats::filter()
x dplyr::lag()     masks stats::lag()</code></pre>
<pre class="r"><code>library(ashr)
library(Matrix)</code></pre>
<pre><code>
Attaching package: &#39;Matrix&#39;</code></pre>
<pre><code>The following objects are masked from &#39;package:tidyr&#39;:

    expand, pack, unpack</code></pre>
<pre class="r"><code>ANT1.2 &lt;- readRDS(&quot;data/ANT1_2.rds&quot;)</code></pre>
</div>
<div id="split-data-into-individuals" class="section level1">
<h1>Split data into individuals</h1>
<p>First, we split the merged seurat object into separate objects corresponding to the individual from which cells originated. Then integrate across the individuals using SCT transform</p>
<pre class="r"><code>#split data into each individual/sample
NA18855_Unstrain &lt;- subset(ANT1.2, labels == &quot;NA18855_Unstrain&quot;)
NA18856_Unstrain &lt;- subset(ANT1.2, labels == &quot;NA18856_Unstrain&quot;)
NA19160_Unstrain &lt;- subset(ANT1.2, labels == &quot;NA19160_Unstrain&quot;)

# Create a seurat object merged for all the unstrain samples (because we are missing one individual for the strained samples)
seurat.list &lt;- list(NA18855_Unstrain, NA18856_Unstrain, NA19160_Unstrain)
for (i in 1:length(seurat.list)) {
    seurat.list[[i]] &lt;- SCTransform(seurat.list[[i]], verbose = FALSE)
}

SCT.features &lt;- SelectIntegrationFeatures(object.list = seurat.list, nfeatures = 5000)
seurat.list &lt;- PrepSCTIntegration(object.list = seurat.list, anchor.features = SCT.features, 
    verbose = FALSE)

#find anchors
seurat.anchors &lt;- FindIntegrationAnchors(object.list = seurat.list, normalization.method = &quot;SCT&quot;, 
    anchor.features = SCT.features, verbose = FALSE)
SCT.integrated &lt;- IntegrateData(anchorset = seurat.anchors, normalization.method = &quot;SCT&quot;,
    verbose = FALSE)

#visualized integrated
SCT.integrated &lt;- RunPCA(SCT.integrated, verbose = FALSE, npcs = 100)
DimPlot(SCT.integrated, reduction = &quot;pca&quot;, group.by = c(&quot;labels&quot;))</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-1">
Past versions of “split single cell data-1.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-1.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ElbowPlot(SCT.integrated, ndims = 100) #38 PCs?</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-2">
Past versions of “split single cell data-2.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-2.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>SCT.integrated &lt;- FindNeighbors(SCT.integrated, dims = 1:38)</code></pre>
<pre><code>Computing nearest neighbor graph</code></pre>
<pre><code>Computing SNN</code></pre>
<pre class="r"><code>SCT.integrated &lt;- FindClusters(SCT.integrated, resolution = 0.4)</code></pre>
<pre><code>Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck

Number of nodes: 1203
Number of edges: 67564

Running Louvain algorithm...
Maximum modularity in 10 random starts: 0.6657
Number of communities: 3
Elapsed time: 0 seconds</code></pre>
<pre class="r"><code>DimPlot(SCT.integrated, group.by = c(&quot;labels&quot;), reduction = &quot;pca&quot;)</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-3">
Past versions of “split single cell data-3.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-3.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>DimPlot(SCT.integrated, group.by = c(&quot;seurat_clusters&quot;), reduction = &quot;pca&quot;)</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-4">
Past versions of “split single cell data-4.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-4.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>SCT.integrated &lt;- RunUMAP(SCT.integrated, dims = 1:38)</code></pre>
<pre><code>18:11:02 UMAP embedding parameters a = 0.9922 b = 1.112</code></pre>
<pre><code>18:11:02 Read 1203 rows and found 38 numeric columns</code></pre>
<pre><code>18:11:02 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>18:11:02 Building Annoy index with metric = cosine, n_trees = 50</code></pre>
<pre><code>0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>[----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>**************************************************|
18:11:03 Writing NN index file to temp file /tmp/RtmpTjEck1/file39d869f3180b
18:11:03 Searching Annoy index using 1 thread, search_k = 3000
18:11:03 Annoy recall = 100%
18:11:03 Commencing smooth kNN distance calibration using 1 thread
18:11:04 Initializing from normalized Laplacian + noise
18:11:04 Commencing optimization for 500 epochs, with 45058 positive edges
18:11:08 Optimization finished</code></pre>
<pre class="r"><code>p1_SCT &lt;- DimPlot(SCT.integrated, group.by = c(&quot;orig.ident&quot;))
p2_SCT &lt;- DimPlot(SCT.integrated, group.by = c(&quot;labels&quot;))
p3_SCT &lt;- DimPlot(SCT.integrated, group.by = c(&quot;seurat_clusters&quot;))
grid.arrange(p1_SCT, p2_SCT, p3_SCT, nrow = 2)</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-5">
Past versions of “split single cell data-5.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-5.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#for figures: remove legend and add in later
p1_figure &lt;- p2_SCT + theme(legend.position = &quot;none&quot;)
p2_figure &lt;- p3_SCT + theme(legend.position = &quot;none&quot;) + scale_color_manual(values=c(&quot;plum3&quot;, &quot;#E69F00&quot;, &quot;#654321&quot;))
grid.arrange(p1_figure, p2_figure)</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-6">
Past versions of “split single cell data-6.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-6.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p2_SCT </code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-7">
Past versions of “split single cell data-7.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-7.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>p3_SCT + scale_color_manual(values=c(&quot;plum3&quot;, &quot;#E69F00&quot;, &quot;#654321&quot;))</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/split%20single%20cell%20data-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-split-single-cell-data-8">
Past versions of “split single cell data-8.png”
</button>
</p>
<div id="fig-no-spaces-split-single-cell-data-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/split single cell data-8.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>SCT.integrated@active.ident &lt;- SCT.integrated$integrated_snn_res.0.4</code></pre>
</div>
<div id="examine-clusters-in-integrated-data" class="section level1">
<h1>Examine clusters in integrated data</h1>
<p>We identified 3 clusters in the integrated data. We now characterize them by gene expression for a chondrogenic marker gene to get a sense of which cluster(s) represent more mature chondrocytes.</p>
<pre class="r"><code>cluster_memberships &lt;- as.matrix(table(SCT.integrated@meta.data$integrated_snn_res.0.4, SCT.integrated@meta.data$labels))
cluster_memberships</code></pre>
<pre><code>   
    NA18855_Unstrain NA18856_Unstrain NA19160_Unstrain
  0              429              280              156
  1              129               81               51
  2               28               34               15</code></pre>
<pre class="r"><code>cluster_memberships_proportions &lt;- prop.table(cluster_memberships, margin = 2)
cluster_memberships_proportions</code></pre>
<pre><code>   
    NA18855_Unstrain NA18856_Unstrain NA19160_Unstrain
  0       0.73208191       0.70886076       0.70270270
  1       0.22013652       0.20506329       0.22972973
  2       0.04778157       0.08607595       0.06756757</code></pre>
<pre class="r"><code>#make a nice proportions bar plot for a figure
long_cluster_memberships_proportions &lt;- as_tibble(as.data.frame(cluster_memberships_proportions))
ggplot(long_cluster_memberships_proportions, aes(x = Var2, y = Freq, fill = Var1)) +
     geom_bar(position = &quot;dodge&quot;, stat = &quot;identity&quot;) + 
     scale_fill_manual(values=c(&quot;plum3&quot;, &quot;#E69F00&quot;, &quot;#654321&quot;)) + 
     theme_cowplot() +
     labs(title = &quot;Proportion of cells from each individual \n assigned to each unsupervised cluster&quot;, x = &quot;&quot;, y = &quot;Proportion&quot;, fill = &quot;Cluster&quot;)</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/examine%20clusters-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-examine-clusters-1">
Past versions of “examine clusters-1.png”
</button>
</p>
<div id="fig-no-spaces-examine-clusters-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/examine clusters-1.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="explore-some-cluster-marker-genes" class="section level1">
<h1>Explore some cluster marker genes</h1>
<p>Here, we fit a poisson ash model to raw counts from each of the three clusters for a few MSC and chondrocyte marker genes and plot their distributions broken down by cluster and by individual. It looks like cluster 2 has higher expression of chondrocyte markers, while more or less all three clusters have similar expression of MSC markers.</p>
<pre class="r"><code>#run poisson ash to data per cluster and per indivdiual, then show the estimated prior distribution
#plot curves for genes separated by indivdual (information for indivdual contained in labels)
plot_by_labels &lt;- function(gene, counts, labels){
     #create dataframe to store the cdf axes and the groups
     dataframe_cdf &lt;- data.frame(x = c(), y = c(), label = c())
     for(individual in unique(labels)){ #in the case that the labels supplied contain cluster, it will do this by cluster instead
          x &lt;- counts[gene, labels == individual]
          s &lt;- colSums(counts[,labels == individual]) #sum of molecule counts per sample
          lam &lt;- x/s
          fit &lt;- ashr::ash_pois(x, s, mixcompdist = &quot;halfuniform&quot;)
          cdf &lt;- ashr::cdf.ash(fit, seq(from = 0, to = max(lam), length.out = 1000))
          temp_df &lt;- cbind(cdf$x, t(cdf$y), individual)
          names(temp_df) &lt;- c(&quot;x&quot;, &quot;y&quot;, &quot;label&quot;)
          dataframe_cdf &lt;- rbind(dataframe_cdf, temp_df)
     }
     names(dataframe_cdf) &lt;- c(&quot;x&quot;, &quot;y&quot;, &quot;label&quot;)
     dataframe_cdf$x &lt;- as.character(dataframe_cdf$x)
     dataframe_cdf$y &lt;- as.character(dataframe_cdf$y)
     dataframe_cdf$x &lt;- as.numeric(dataframe_cdf$x)
     dataframe_cdf$y &lt;- as.numeric(dataframe_cdf$y)
     #use df to make our plot
     plot &lt;- ggplot(dataframe_cdf, aes(x = x, y = y, color = as.factor(label), group=as.factor(label))) +
          geom_point() + 
          labs(title = paste0(gene), x = &quot;Latent gene expression&quot;, y = &quot;CDF&quot;, color = &quot;Label&quot;) +
          theme_cowplot()
     return(plot)
}

features.plot.chond &lt;- c(&quot;TIMP3&quot;, &quot;TIMP2&quot;, &quot;COL11A1&quot;, 
                   &quot;SOX5&quot;, &quot;SOX6&quot;,
                   &quot;TGFBI&quot;)
features.plot.msc &lt;- c(&quot;THY1&quot;,  &quot;NT5E&quot;)

#by cluster
for (marker in c(features.plot.msc, features.plot.chond)){
     print(plot_by_labels(marker, SCT.integrated@assays$RNA@counts, SCT.integrated$integrated_snn_res.0.4) + labs(color = &quot;Cluster&quot;))
}</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-1">
Past versions of “poisson ash-1.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-1.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-2">
Past versions of “poisson ash-2.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-2.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-3">
Past versions of “poisson ash-3.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-3.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-4">
Past versions of “poisson ash-4.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-4.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-5">
Past versions of “poisson ash-5.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-5.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-6">
Past versions of “poisson ash-6.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-6.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-7">
Past versions of “poisson ash-7.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-7.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-8">
Past versions of “poisson ash-8.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-8.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#by individual
for (marker in c(features.plot.msc, features.plot.chond)){
     print(plot_by_labels(marker, SCT.integrated@assays$RNA@counts, SCT.integrated$labels) + labs(color = &quot;Individual&quot;))
}</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-9">
Past versions of “poisson ash-9.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-9.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-10">
Past versions of “poisson ash-10.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-10.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-11.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-11">
Past versions of “poisson ash-11.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-11" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-11.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-12.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-12">
Past versions of “poisson ash-12.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-12" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-12.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-13.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-13">
Past versions of “poisson ash-13.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-13" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-13.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-14.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-14">
Past versions of “poisson ash-14.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-14" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-14.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-15.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-15">
Past versions of “poisson ash-15.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-15" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-15.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-16.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-16">
Past versions of “poisson ash-16.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-16" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-16.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#plots for figure
a &lt;- plot_by_labels(&quot;COL11A1&quot;, SCT.integrated@assays$RNA@counts, SCT.integrated$integrated_snn_res.0.4) + 
     theme(legend.position = &quot;none&quot;) + 
     scale_color_manual(values=c(&quot;plum3&quot;, &quot;#E69F00&quot;, &quot;#654321&quot;))
b &lt;- plot_by_labels(&quot;COL11A1&quot;, SCT.integrated@assays$RNA@counts, SCT.integrated$labels) + 
     theme(legend.position = &quot;none&quot;) 
grid.arrange(a, b, nrow = 1)</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-17.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-17">
Past versions of “poisson ash-17.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-17" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-17.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>a&lt;- c(&quot;MMP2&quot;, &quot;MMP14&quot;)

for (marker in a){
     print(plot_by_labels(marker, SCT.integrated@assays$RNA@counts, SCT.integrated$integrated_snn_res.0.4) + labs(color = &quot;Cluster&quot;))
}</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-18.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-18">
Past versions of “poisson ash-18.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-18" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-18.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-19.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-19">
Past versions of “poisson ash-19.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-19" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-19.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#by individual
for (marker in a){
     print(plot_by_labels(marker, SCT.integrated@assays$RNA@counts, SCT.integrated$labels) + labs(color = &quot;Individual&quot;))
}</code></pre>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-20.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-20">
Past versions of “poisson ash-20.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-20" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-20.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/clustering_scRNA.Rmd/poisson%20ash-21.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-poisson-ash-21">
Past versions of “poisson ash-21.png”
</button>
</p>
<div id="fig-no-spaces-poisson-ash-21" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/9cf26dc79b2f56a205b44ad6c67f9b8d7d52739c/docs/figure/clustering_scRNA.Rmd/poisson ash-21.png" target="_blank">9cf26dc</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Matrix_1.2-18   ashr_2.2-47     forcats_0.4.0   stringr_1.4.0  
 [5] dplyr_1.0.2     purrr_0.3.4     readr_1.3.1     tidyr_1.1.2    
 [9] tibble_3.0.4    tidyverse_1.3.0 cowplot_1.1.0   ggplot2_3.3.3  
[13] gridExtra_2.3   Seurat_3.2.3   

loaded via a namespace (and not attached):
  [1] readxl_1.3.1          backports_1.1.10      workflowr_1.6.2      
  [4] plyr_1.8.6            igraph_1.2.4.1        lazyeval_0.2.2       
  [7] splines_3.6.1         listenv_0.8.0         scattermore_0.7      
 [10] digest_0.6.27         invgamma_1.1          htmltools_0.5.0      
 [13] SQUAREM_2020.4        gdata_2.18.0          fansi_0.4.1          
 [16] magrittr_2.0.1        tensor_1.5            cluster_2.1.0        
 [19] ROCR_1.0-7            globals_0.12.5        modelr_0.1.8         
 [22] matrixStats_0.57.0    colorspace_2.0-0      rvest_0.3.6          
 [25] rappdirs_0.3.1        ggrepel_0.9.0         haven_2.3.1          
 [28] xfun_0.8              crayon_1.3.4          jsonlite_1.7.2       
 [31] spatstat_1.64-1       spatstat.data_1.7-0   survival_2.44-1.1    
 [34] zoo_1.8-8             glue_1.4.2            polyclip_1.10-0      
 [37] gtable_0.3.0          leiden_0.3.1          future.apply_1.3.0   
 [40] abind_1.4-5           scales_1.1.1          DBI_1.1.0            
 [43] miniUI_0.1.1.1        Rcpp_1.0.5            viridisLite_0.3.0    
 [46] xtable_1.8-4          reticulate_1.16       rsvd_1.0.1           
 [49] truncnorm_1.0-8       htmlwidgets_1.5.2     httr_1.4.2           
 [52] gplots_3.0.1.1        RColorBrewer_1.1-2    ellipsis_0.3.1       
 [55] ica_1.0-2             farver_2.0.3          pkgconfig_2.0.3      
 [58] uwot_0.1.10           dbplyr_1.4.2          deldir_0.1-23        
 [61] labeling_0.4.2        tidyselect_1.1.0      rlang_0.4.10         
 [64] reshape2_1.4.3        later_1.1.0.1         munsell_0.5.0        
 [67] cellranger_1.1.0      tools_3.6.1           cli_2.2.0            
 [70] generics_0.0.2        broom_0.7.0           ggridges_0.5.1       
 [73] evaluate_0.14         yaml_2.2.1            goftest_1.2-2        
 [76] npsurv_0.4-0          knitr_1.23            fs_1.3.1             
 [79] fitdistrplus_1.0-14   caTools_1.17.1.2      RANN_2.6.1           
 [82] pbapply_1.4-0         future_1.18.0         nlme_3.1-140         
 [85] whisker_0.3-2         mime_0.9              xml2_1.3.2           
 [88] compiler_3.6.1        rstudioapi_0.13       plotly_4.9.2.1       
 [91] png_0.1-7             lsei_1.2-0            spatstat.utils_1.17-0
 [94] reprex_0.3.0          stringi_1.4.6         RSpectra_0.15-0      
 [97] lattice_0.20-41       vctrs_0.3.6           pillar_1.4.7         
[100] lifecycle_0.2.0       lmtest_0.9-37         RcppAnnoy_0.0.18     
[103] data.table_1.13.0     bitops_1.0-6          irlba_2.3.3          
[106] httpuv_1.5.1          patchwork_1.1.0       R6_2.5.0             
[109] promises_1.1.1        KernSmooth_2.23-15    codetools_0.2-16     
[112] MASS_7.3-52           gtools_3.8.1          assertthat_0.2.1     
[115] rprojroot_2.0.2       withr_2.3.0           sctransform_0.3.2    
[118] mgcv_1.8-28           parallel_3.6.1        hms_0.5.3            
[121] grid_3.6.1            rpart_4.1-15          rmarkdown_1.13       
[124] Rtsne_0.15            git2r_0.26.1          mixsqp_0.3-43        
[127] shiny_1.3.2           lubridate_1.7.9      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
