<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Anthony Hung" />

<meta name="date" content="2021-01-19" />

<title>Aggregation of external scRNA-seq data for topic model</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->



<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">invitroOA_pilot_repository</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Aggregation of external scRNA-seq data for topic model</h1>
<h4 class="author">Anthony Hung</h4>
<h4 class="date">2021-01-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-20
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>invitroOA_pilot_repository/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree99c70b8e43338b294c4aa7877ea7eccc732be2actargetblank99c70b8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/99c70b8e43338b294c4aa7877ea7eccc732be2ac" target="_blank">99c70b8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree99c70b8e43338b294c4aa7877ea7eccc732be2actargetblank99c70b8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/99c70b8e43338b294c4aa7877ea7eccc732be2ac" target="_blank">99c70b8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/conda-archive/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/conda/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/locks/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/shadow/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/singularity/
    Ignored:    code/bulkRNA_preprocessing/.snakemake/tmp.3ekfs3n5/
    Ignored:    code/bulkRNA_preprocessing/fastq/
    Ignored:    code/single_cell_preprocessing/.snakemake/conda-archive/
    Ignored:    code/single_cell_preprocessing/.snakemake/conda/
    Ignored:    code/single_cell_preprocessing/.snakemake/locks/
    Ignored:    code/single_cell_preprocessing/.snakemake/shadow/
    Ignored:    code/single_cell_preprocessing/.snakemake/singularity/
    Ignored:    code/single_cell_preprocessing/YG-AH-2S-ANT-1_S1_L008/
    Ignored:    code/single_cell_preprocessing/YG-AH-2S-ANT-2_S2_L008/
    Ignored:    code/single_cell_preprocessing/demuxlet/.DS_Store
    Ignored:    code/single_cell_preprocessing/fastq/
    Ignored:    data/ANT1_2.rds
    Ignored:    data/external_scRNA/Chou_et_al2020/
    Ignored:    data/external_scRNA/Jietal2018/
    Ignored:    data/external_scRNA/Wuetal2021/
    Ignored:    data/external_scRNA/merged_external_scRNA.rds

Unstaged changes:
    Modified:   analysis/powerAnalysis.Rmd
    Modified:   analysis/topicModel_scRNA.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/external_scRNA.Rmd</code>) and HTML (<code>docs/external_scRNA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/99c70b8e43338b294c4aa7877ea7eccc732be2ac/analysis/external_scRNA.Rmd" target="_blank">99c70b8</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
update external data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/99c70b8e43338b294c4aa7877ea7eccc732be2ac/docs/external_scRNA.html" target="_blank">99c70b8</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
update external data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/1ec66ea82cb6f4c6fc39b238c58fca311f44cf64/analysis/external_scRNA.Rmd" target="_blank">1ec66ea</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add information desribing preprocessing scran
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/1ec66ea82cb6f4c6fc39b238c58fca311f44cf64/docs/external_scRNA.html" target="_blank">1ec66ea</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add information desribing preprocessing scran
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/af83ab7bab9d7a1b7dabf1dc65e2eea7f94833de/analysis/external_scRNA.Rmd" target="_blank">af83ab7</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
knit analysis files
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/af83ab7bab9d7a1b7dabf1dc65e2eea7f94833de/docs/external_scRNA.html" target="_blank">af83ab7</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
knit analysis files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/98e0a1d600c1e4717c2028b2a6ddd3ba84f41426/analysis/external_scRNA.Rmd" target="_blank">98e0a1d</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add note about running the bash code separately
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/5b2d20e585d757ffc6af013209f292e555dcb73a/analysis/external_scRNA.Rmd" target="_blank">5b2d20e</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
Commit before knitting
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/439d1945c0159f33d8e26066a0f76499006c17b8/analysis/external_scRNA.Rmd" target="_blank">439d194</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add liver data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/c49176b561d2c7a9e3ddfabc37020712b99cb3ad/analysis/external_scRNA.Rmd" target="_blank">c49176b</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
add GH data into external datasets folder
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/8b39783546b928e0cb83fdcbfe3c87477c5f9161/analysis/external_scRNA.Rmd" target="_blank">8b39783</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
correct typo in directory structure creation in external scRNAseq
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e24a0cd0423eec17120b96334a3cac2c63eef3e8/analysis/external_scRNA.Rmd" target="_blank">e24a0cd</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
Added download links for external data files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/d6b5b17900d3314edb102b088ef4a7b12cf2a6ca/analysis/external_scRNA.Rmd" target="_blank">d6b5b17</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-20
</td>
<td>
Added analysis files
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/28f57faed7255644b4858290aed125d5466312ea/analysis/external_scRNA.Rmd" target="_blank">28f57fa</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-19
</td>
<td>
Add files for analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-in-datasets-for-topic-model-and-merge-them-into-one-seurat-object" class="section level1">
<h1>Load in datasets for topic model and merge them into one seurat object</h1>
</div>
<div id="download-datasets-from-geo" class="section level1">
<h1>Download datasets from GEO</h1>
<p>Some of the data included in this model come from publically available datasets from published studies, housed in GEO. Run the following code from the home directory of this repo to download the necessary datasets.</p>
<pre class="bash"><code>mkdir data/external_scRNA
cd data/external_scRNA

#download Ji et al 2018 data of Chondrocytes from OA patients run through STRT single cell RNA protocol from GEO (Series GSE104782)
mkdir Jietal2018
cd Jietal2018
wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE104nnn/GSE104782/suppl/GSE104782_allcells_UMI_count.txt.gz #raw count data
wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE104nnn/GSE104782/suppl/GSE104782_Table_Cell_quality_information_and_clustering_information.xlsx # clustering information for cells in the raw data

#download Wu et al 2021 data of iPSC-Chondrocyte chondrogenic pellet differentiation time course from GEO (Series GSE160625). Data include barcodes, genes, and count matrices from samples taken from iPSCs and samples from 7, 14, 28, and 42 day chondrogenic pellets treated with chondrogenic media containing a WNT and MITF inhibitor
cd ..
mkdir Wuetal2021
cd Wuetal2021
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876128/suppl/GSM4876128_hiPSC_barcodes.tsv.gz # iPSC barcodes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876128/suppl/GSM4876128_hiPSC_genes.tsv.gz # iPSC genes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876128/suppl/GSM4876128_hiPSC_matrix.mtx.gz # iPSC count matrix
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876136/suppl/GSM4876136_C59_D7_barcodes.tsv.gz # day 7 pellet barcodes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876136/suppl/GSM4876136_C59_D7_genes.tsv.gz # day 7 pellet genes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876136/suppl/GSM4876136_C59_D7_matrix.mtx.gz # day 7 pellet count matrix
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876137/suppl/GSM4876137_C59_D14_barcodes.tsv.gz # day 14 pellet barcodes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876137/suppl/GSM4876137_C59_D14_genes.tsv.gz # day 14 pellet genes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876137/suppl/GSM4876137_C59_D14_matrix.mtx.gz  # day 14 pellet count matrix
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876138/suppl/GSM4876138_C59_D28_barcodes.tsv.gz # day 28 pellet barcodes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876138/suppl/GSM4876138_C59_D28_genes.tsv.gz # day 28 pellet genes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876138/suppl/GSM4876138_C59_D28_matrix.mtx.gz # day 28 pellet count matrix
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876139/suppl/GSM4876139_C59_D42_barcodes.tsv.gz # day 42 pellet barcodes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876139/suppl/GSM4876139_C59_D42_genes.tsv.gz # day 42 pellet genes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4876nnn/GSM4876139/suppl/GSM4876139_C59_D42_matrix.mtx.gz # day 42 pellet count matrix

#download data for Chou et al 2020 from GEO (Sample GSM4626766) for chondrocytes from Minimally damaged outer lateral tibial (oLT) plateau cartilage isolated from one male individual
cd ..
mkdir Chou_et_al2020
cd Chou_et_al2020
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4626nnn/GSM4626766/suppl/GSM4626766_OA_oLT_113.barcodes.tsv.gz # barcodes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4626nnn/GSM4626766/suppl/GSM4626766_OA_oLT_113.genes.tsv.gz # genes
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM4626nnn/GSM4626766/suppl/GSM4626766_OA_oLT_113.matrix.mtx.gz # count matrix</code></pre>
</div>
<div id="load-in-datasets" class="section level1">
<h1>Load in datasets</h1>
<p>Load in the datasets and create Seurat objects for each of them. Add metadata about the cell type they were assigned in their original contexts. Merge the Seurat object into one large object.</p>
<pre class="r"><code>library(Matrix)
library(Seurat)
library(tidyverse)</code></pre>
<pre><code>Registered S3 method overwritten by &#39;cli&#39;:
  method     from    
  print.boxx spatstat</code></pre>
<pre><code>── Attaching packages ────────────────────────────────── tidyverse 1.3.0 ──</code></pre>
<pre><code>✓ ggplot2 3.3.3     ✓ purrr   0.3.4
✓ tibble  3.0.4     ✓ dplyr   1.0.2
✓ tidyr   1.1.2     ✓ stringr 1.4.0
✓ readr   1.3.1     ✓ forcats 0.4.0</code></pre>
<pre><code>── Conflicts ───────────────────────────────────── tidyverse_conflicts() ──
x tidyr::expand() masks Matrix::expand()
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
x tidyr::pack()   masks Matrix::pack()
x tidyr::unpack() masks Matrix::unpack()</code></pre>
<pre class="r"><code>library(readxl)

#iPSC-Chondrocyte data from the current study
ANT1.2 &lt;- readRDS(&quot;data/ANT1_2.rds&quot;)
ANT1.2 &lt;- Seurat::AddMetaData(ANT1.2, &quot;iPSC-Chondrocyte&quot;, col.name = &quot;Cell.Type&quot;)
dim(ANT1.2)</code></pre>
<pre><code>[1] 33538  1815</code></pre>
<pre class="r"><code>#Chondrocytes from OA patients (https://www.ncbi.nlm.nih.gov/pubmed/30026257) run through STRT protocol
# load raw count data
ji_chondro &lt;- read.table(&quot;data/external_scRNA/Jietal2018/GSE104782_allcells_UMI_count.txt.gz&quot;, header = T)
ji_chondro_genes &lt;- ji_chondro[,1]
ji_chondro_data &lt;- ji_chondro[,-1]
ji_chondro &lt;- ji_chondro_data
rownames(ji_chondro) &lt;- ji_chondro_genes
# load clustering assignment data from paper
ji_chondro_metadata &lt;- read_excel(&quot;data/external_scRNA/Jietal2018/GSE104782_Table_Cell_quality_information_and_clustering_information.xlsx&quot;)
# subset count data to include only those cells that were assigned to a cluster labeled as being a chondrocyte in the original paper
ji_chondro.filtered &lt;- ji_chondro[,!is.na(ji_chondro_metadata$Cluster)]
cluster.ident &lt;- &quot;ji_chondrocyte&quot;
# Create seurat object from chondrocytes data
ji_chondro.seurat &lt;- CreateSeuratObject(ji_chondro.filtered, project=&quot;Chondrocytes&quot;)</code></pre>
<pre><code>Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>ji_chondro.seurat &lt;- Seurat::AddMetaData(ji_chondro.seurat, cluster.ident, col.name = &quot;Cell.Type&quot;)
dim(ji_chondro.seurat)</code></pre>
<pre><code>[1] 24153  1464</code></pre>
<pre class="r"><code>#iPSC chondrogenic cells from directed differentiation time course  from iPSC chondrogenic pellet culture. https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE160625 
count_data_wu_d7 &lt;- readMM(&quot;data/external_scRNA/Wuetal2021/GSM4876136_C59_D7_matrix.mtx.gz&quot;)
genes_wu_d7 &lt;- read_tsv(&quot;data/external_scRNA/Wuetal2021/GSM4876136_C59_D7_genes.tsv.gz&quot;, col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>barcodes_wu_d7 &lt;- as.data.frame(read_tsv( &quot;data/external_scRNA/Wuetal2021/GSM4876136_C59_D7_barcodes.tsv.gz&quot;, col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code>rownames(count_data_wu_d7) &lt;- genes_wu_d7$X2
colnames(count_data_wu_d7) &lt;- barcodes_wu_d7$X1
wu_d7_seurat &lt;- CreateSeuratObject(counts = count_data_wu_d7, project = &quot;d7&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique

Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>wu_d7_seurat &lt;- Seurat::AddMetaData(wu_d7_seurat, &quot;Wu_chondrogenic_pellet_d7&quot;, col.name = &quot;Cell.Type&quot;)

count_data_wu_d14 &lt;- readMM(&quot;data/external_scRNA/Wuetal2021/GSM4876137_C59_D14_matrix.mtx.gz&quot;)
genes_wu_d14 &lt;- read_tsv(&quot;data/external_scRNA/Wuetal2021/GSM4876137_C59_D14_genes.tsv.gz&quot;, col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>barcodes_wu_d14 &lt;- as.data.frame(read_tsv( &quot;data/external_scRNA/Wuetal2021/GSM4876137_C59_D14_barcodes.tsv.gz&quot;, col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code>rownames(count_data_wu_d14) &lt;- genes_wu_d14$X2
colnames(count_data_wu_d14) &lt;- barcodes_wu_d14$X1
wu_d14_seurat &lt;- CreateSeuratObject(counts = count_data_wu_d14, project = &quot;d14&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique

Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>wu_d14_seurat &lt;- Seurat::AddMetaData(wu_d14_seurat, &quot;Wu_chondrogenic_pellet_d14&quot;, col.name = &quot;Cell.Type&quot;)

count_data_wu_d28 &lt;- readMM(&quot;data/external_scRNA/Wuetal2021/GSM4876138_C59_D28_matrix.mtx.gz&quot;)
genes_wu_d28 &lt;- read_tsv(&quot;data/external_scRNA/Wuetal2021/GSM4876138_C59_D28_genes.tsv.gz&quot;, col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>barcodes_wu_d28 &lt;- as.data.frame(read_tsv( &quot;data/external_scRNA/Wuetal2021/GSM4876138_C59_D28_barcodes.tsv.gz&quot;, col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code>rownames(count_data_wu_d28) &lt;- genes_wu_d28$X2
colnames(count_data_wu_d28) &lt;- barcodes_wu_d28$X1
wu_d28_seurat &lt;- CreateSeuratObject(counts = count_data_wu_d28, project = &quot;d28&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique

Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>wu_d28_seurat &lt;- Seurat::AddMetaData(wu_d28_seurat, &quot;Wu_chondrogenic_pellet_d28&quot;, col.name = &quot;Cell.Type&quot;)

count_data_wu_d42 &lt;- readMM(&quot;data/external_scRNA/Wuetal2021/GSM4876139_C59_D42_matrix.mtx.gz&quot;)
genes_wu_d42 &lt;- read_tsv(&quot;data/external_scRNA/Wuetal2021/GSM4876139_C59_D42_genes.tsv.gz&quot;, col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>barcodes_wu_d42 &lt;- as.data.frame(read_tsv( &quot;data/external_scRNA/Wuetal2021/GSM4876139_C59_D42_barcodes.tsv.gz&quot;, col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code>rownames(count_data_wu_d42) &lt;- genes_wu_d42$X2
colnames(count_data_wu_d42) &lt;- barcodes_wu_d42$X1
wu_d42_seurat &lt;- CreateSeuratObject(counts = count_data_wu_d42, project = &quot;d42&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique

Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>wu_d42_seurat &lt;- Seurat::AddMetaData(wu_d42_seurat, &quot;Wu_chondrogenic_pellet_d42&quot;, col.name = &quot;Cell.Type&quot;)

wu_combined &lt;- merge(wu_d7_seurat, y = c(wu_d14_seurat, wu_d28_seurat, wu_d42_seurat), project = &quot;Combined.common&quot;)</code></pre>
<pre><code>Warning in CheckDuplicateCellNames(object.list = objects): Some cell names
are duplicated across objects provided. Renaming to enforce unique cell
names.</code></pre>
<pre class="r"><code>wu_combined[[&quot;percent.mt&quot;]] &lt;- PercentageFeatureSet(wu_combined, pattern = &quot;^MT-&quot;)
wu_combined &lt;- subset(wu_combined, subset = percent.mt &lt; 25)
dim(wu_combined)</code></pre>
<pre><code>[1] 45924  9434</code></pre>
<pre class="r"><code>#Chondrocytes from healthy oLT of a male human with OA (outer lateral tibia) https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4626766
chou_chondro_counts &lt;- readMM(&quot;data/external_scRNA/Chou_et_al2020/GSM4626766_OA_oLT_113.matrix.mtx.gz&quot;)
chou_chondro_genes &lt;- read_tsv(&quot;data/external_scRNA/Chou_et_al2020/GSM4626766_OA_oLT_113.genes.tsv.gz&quot;, col_names = F)</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character(),
  X2 = col_character()
)</code></pre>
<pre class="r"><code>chou_chondro_barcodes &lt;- as.data.frame(read_tsv(&quot;data/external_scRNA/Chou_et_al2020/GSM4626766_OA_oLT_113.barcodes.tsv.gz&quot;, col_names = F))</code></pre>
<pre><code>Parsed with column specification:
cols(
  X1 = col_character()
)</code></pre>
<pre class="r"><code>#make Seurat object
rownames(chou_chondro_counts) &lt;- chou_chondro_genes$X2
colnames(chou_chondro_counts) &lt;- chou_chondro_barcodes$X1

chou_chondro_seurat &lt;- CreateSeuratObject(counts = chou_chondro_counts, project = &quot;Chondrocytes&quot;) %&gt;% 
  AddMetaData(&quot;chou_chondrocyte&quot;, col.name = &quot;Cell.Type&quot;)</code></pre>
<pre><code>Warning: Non-unique features (rownames) present in the input matrix, making
unique

Warning: Feature names cannot have underscores (&#39;_&#39;), replacing with dashes
(&#39;-&#39;)</code></pre>
<pre class="r"><code>dim(chou_chondro_seurat)</code></pre>
<pre><code>[1] 32738  6200</code></pre>
<pre class="r"><code>#iPSCs/MSCs/ipsc-chondrocytes/ipsc-osteoblasts from Genevieve Housman. Combine all the datasets located in the data directory. Each of these datasets contain single cell data from cells from the same human individual from which iPSCs were created. 10x single cell RNA sequencing data from iPSCs, MSCs, iPSC-Osteoblasts, and iPSC-chondrocytes are included after filtering for barcodes with fewer than 25% reads coming from mitochondria.
GH_iPSC &lt;- readRDS(&quot;data/external_scRNA/GH_cells/gh_ipsc.rds&quot;)
GH_iPSCMSC &lt;- readRDS(&quot;data/external_scRNA/GH_cells/gh_msc.rds&quot;)
GH_iPSCChond &lt;- readRDS(&quot;data/external_scRNA/GH_cells/gh_ipscChond.rds&quot;)
GH_iPSCOsteo &lt;- readRDS(&quot;data/external_scRNA/GH_cells/gh_ipscOsteo.rds&quot;)
#merge all the datasets into one seurat object
GH_merged &lt;- merge(GH_iPSC, y = c(GH_iPSCMSC, GH_iPSCChond, GH_iPSCOsteo), 
                add.cell.ids = c(&quot;ipsc&quot;, &quot;msc&quot;, &quot;chondrogh&quot;, &quot;osteo&quot;), project = &quot;ghousman&quot;)
GH_merged &lt;- Seurat::AddMetaData(GH_merged, &quot;Ghousman&quot;, col.name = &quot;orig.ident&quot;)
dim(GH_merged)</code></pre>
<pre><code>[1] 19377 11146</code></pre>
<pre class="r"><code>#Human Liver samples run through 10x (https://doi.org/10.1038/s41467-018-06318-7). This is a subset of the entire dataset, which is from whole liver homogenate. Following the example from the original paper, scViz was used to subset the data for clusters that were called hepatocytes in the original study (subset contains a total of 3490 cells)
liver_data &lt;- readRDS(&quot;data/external_scRNA/humanLiverSubset_hepatocytes.rds&quot;)
liver_data &lt;- Seurat::AddMetaData(liver_data, &quot;Hepatocyte&quot;, col.name = &quot;Cell.Type&quot;)
dim(liver_data)</code></pre>
<pre><code>[1] 20007  3490</code></pre>
</div>
<div id="merge-and-save-the-combined-data" class="section level1">
<h1>Merge and Save the combined data</h1>
<pre class="r"><code># Merge Datasets

# Merge the seurat objects into one large one, keeping metadata about which study they originated in as well as their assigned cell type.
Merged_external_data &lt;- merge(ANT1.2, y = c(ji_chondro.seurat, chou_chondro_seurat, GH_merged, liver_data), project = &quot;Merged_external&quot;)
Merged_external_data &lt;- merge(Merged_external_data, wu_combined, project = &quot;Merged_external&quot;)
dim(Merged_external_data)</code></pre>
<pre><code>[1] 61987 33549</code></pre>
<pre class="r"><code>saveRDS(Merged_external_data, &quot;data/external_scRNA/merged_external_scRNA.rds&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.1 (2019-07-05)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Scientific Linux 7.4 (Nitrogen)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.2.19-el7-x86_64/lib/libopenblas_haswellp-r0.2.19.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1    forcats_0.4.0   stringr_1.4.0   dplyr_1.0.2    
 [5] purrr_0.3.4     readr_1.3.1     tidyr_1.1.2     tibble_3.0.4   
 [9] ggplot2_3.3.3   tidyverse_1.3.0 Seurat_3.2.3    Matrix_1.2-18  

loaded via a namespace (and not attached):
  [1] Rtsne_0.15            colorspace_2.0-0      deldir_0.1-23        
  [4] ellipsis_0.3.1        ggridges_0.5.1        rprojroot_2.0.2      
  [7] fs_1.3.1              rstudioapi_0.13       spatstat.data_1.7-0  
 [10] leiden_0.3.1          listenv_0.8.0         npsurv_0.4-0         
 [13] ggrepel_0.9.0         fansi_0.4.1           lubridate_1.7.9      
 [16] xml2_1.3.2            codetools_0.2-16      splines_3.6.1        
 [19] lsei_1.2-0            knitr_1.23            polyclip_1.10-0      
 [22] jsonlite_1.7.2        workflowr_1.6.2       broom_0.7.0          
 [25] ica_1.0-2             dbplyr_1.4.2          cluster_2.1.0        
 [28] png_0.1-7             uwot_0.1.10           shiny_1.3.2          
 [31] sctransform_0.3.2     compiler_3.6.1        httr_1.4.2           
 [34] backports_1.1.10      assertthat_0.2.1      lazyeval_0.2.2       
 [37] cli_2.2.0             later_1.1.0.1         htmltools_0.5.0      
 [40] tools_3.6.1           rsvd_1.0.1            igraph_1.2.4.1       
 [43] gtable_0.3.0          glue_1.4.2            RANN_2.6.1           
 [46] reshape2_1.4.3        rappdirs_0.3.1        Rcpp_1.0.5           
 [49] spatstat_1.64-1       scattermore_0.7       cellranger_1.1.0     
 [52] vctrs_0.3.6           gdata_2.18.0          nlme_3.1-140         
 [55] lmtest_0.9-37         xfun_0.8              globals_0.12.5       
 [58] rvest_0.3.6           mime_0.9              miniUI_0.1.1.1       
 [61] lifecycle_0.2.0       irlba_2.3.3           gtools_3.8.1         
 [64] goftest_1.2-2         future_1.18.0         MASS_7.3-52          
 [67] zoo_1.8-8             scales_1.1.1          hms_0.5.3            
 [70] promises_1.1.1        spatstat.utils_1.17-0 parallel_3.6.1       
 [73] RColorBrewer_1.1-2    yaml_2.2.1            reticulate_1.16      
 [76] pbapply_1.4-0         gridExtra_2.3         rpart_4.1-15         
 [79] stringi_1.4.6         caTools_1.17.1.2      rlang_0.4.10         
 [82] pkgconfig_2.0.3       matrixStats_0.57.0    bitops_1.0-6         
 [85] evaluate_0.14         lattice_0.20-41       ROCR_1.0-7           
 [88] tensor_1.5            patchwork_1.1.0       htmlwidgets_1.5.2    
 [91] cowplot_1.1.0         tidyselect_1.1.0      RcppAnnoy_0.0.18     
 [94] plyr_1.8.6            magrittr_2.0.1        R6_2.5.0             
 [97] gplots_3.0.1.1        generics_0.0.2        DBI_1.1.0            
[100] withr_2.3.0           haven_2.3.1           pillar_1.4.7         
[103] whisker_0.3-2         mgcv_1.8-28           fitdistrplus_1.0-14  
[106] survival_2.44-1.1     abind_1.4-5           future.apply_1.3.0   
[109] modelr_0.1.8          crayon_1.3.4          KernSmooth_2.23-15   
[112] plotly_4.9.2.1        rmarkdown_1.13        grid_3.6.1           
[115] data.table_1.13.0     git2r_0.26.1          reprex_0.3.0         
[118] digest_0.6.27         xtable_1.8-4          httpuv_1.5.1         
[121] munsell_0.5.0         viridisLite_0.3.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
