<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Anthony Hung" />

<meta name="date" content="2021-01-19" />

<title>Normalization and filtering of bulkRNAseq data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">invitroOA_pilot_repository</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Normalization and filtering of bulkRNAseq data</h1>
<h4 class="author">Anthony Hung</h4>
<h4 class="date">2021-01-19</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-01-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>invitroOA_pilot_repository/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210119code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210119)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210119code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210119)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree78cfbcd3204249a8673228d684ae77444c63266btargetblank78cfbcda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/78cfbcd3204249a8673228d684ae77444c63266b" target="_blank">78cfbcd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomanthonyhunginvitroOApilotrepositorytree78cfbcd3204249a8673228d684ae77444c63266btargetblank78cfbcda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/anthonyhung/invitroOA_pilot_repository/tree/78cfbcd3204249a8673228d684ae77444c63266b" target="_blank">78cfbcd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    data/external_scRNA/.DS_Store

Unstaged changes:
    Deleted:    analysis/figure/powerAnalysis.Rmd/curve-1.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-1.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-2.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-3.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-4.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-5.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-6.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-7.png
    Deleted:    analysis/figure/powerAnalysis.Rmd/hypoxia-8.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/normFilter_bulkRNA.Rmd</code>) and HTML (<code>docs/normFilter_bulkRNA.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/analysis/normFilter_bulkRNA.Rmd" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
add ward et al data files to data
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/normFilter_bulkRNA.html" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
add ward et al data files to data
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/analysis/normFilter_bulkRNA.Rmd" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
start normalization file
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/anthonyhung/invitroOA_pilot_repository/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/normFilter_bulkRNA.html" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
<td>
start normalization file
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/28f57faed7255644b4858290aed125d5466312ea/analysis/normFilter_bulkRNA.Rmd" target="_blank">28f57fa</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-19
</td>
<td>
Add files for analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>This code takes the raw count data from bulk RNA sequencing and performs upper quartile normalziation on the data, filtering out lowly expressed genes. It then saves the resulting data and filtered count data (unnormalized) as R objects.</p>
<p>Here we also plot PCA plots to visualize the structure of the full data set (PCA plots for the data set excluding the sample that failed QC are located in the <a href="removeUnwantedVariation_bulkRNA.html">Modeling and removing unwanted technical variation</a> page).</p>
</div>
<div id="load-data-and-libraries" class="section level1">
<h1>Load data and libraries</h1>
<p>Here we are loading both the full data and the data without the one sample that failed QC.</p>
<pre class="r"><code>library(&quot;gplots&quot;)</code></pre>
<pre><code>
Attaching package: &#39;gplots&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:stats&#39;:

    lowess</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)
library(&quot;reshape&quot;)
library(&quot;edgeR&quot;)</code></pre>
<pre><code>Loading required package: limma</code></pre>
<pre class="r"><code>library(&quot;RColorBrewer&quot;)
library(&quot;scales&quot;)
library(&quot;cowplot&quot;)</code></pre>
<pre><code>
Attaching package: &#39;cowplot&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:reshape&#39;:

    stamp</code></pre>
<pre class="r"><code>theme_set(theme_cowplot())
library(&quot;dplyr&quot;)</code></pre>
<pre><code>
Attaching package: &#39;dplyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:reshape&#39;:

    rename</code></pre>
<pre><code>The following objects are masked from &#39;package:stats&#39;:

    filter, lag</code></pre>
<pre><code>The following objects are masked from &#39;package:base&#39;:

    intersect, setdiff, setequal, union</code></pre>
<pre class="r"><code># Load colors 
colors &lt;- colorRampPalette(c(brewer.pal(9, &quot;Blues&quot;)[1],brewer.pal(9, &quot;Blues&quot;)[9]))(100)
pal &lt;- c(brewer.pal(9, &quot;Set1&quot;), brewer.pal(8, &quot;Set2&quot;), brewer.pal(12, &quot;Set3&quot;))


# load in relabeled counts
raw_counts &lt;- readRDS(&quot;data/raw_counts_relabeled.rds&quot;)
raw_counts_includeNA19160 &lt;- readRDS(&quot;data/raw_counts_relabeled_includeNA19160.rds&quot;)

# Create DGEList object to allow for easier application of different normalization methods
raw_counts &lt;- DGEList(raw_counts, group = colnames(raw_counts))
raw_counts_includeNA19160 &lt;- DGEList(raw_counts_includeNA19160, group = colnames(raw_counts_includeNA19160))

# load in reordered sample information
sampleinfo &lt;- readRDS(&quot;data/Sample.info.RNAseq.reordered.rds&quot;)
sampleinfo_includeNA19160 &lt;- readRDS(&quot;data/Sample.info.RNAseq.reordered_includeNA19160.rds&quot;)</code></pre>
</div>
<div id="upperquartile-normalization" class="section level1">
<h1>Upperquartile Normalization</h1>
<p>Apply upperquartile normalization to the data.</p>
<pre class="r"><code>upperquartile &lt;- calcNormFactors(raw_counts, method = &quot;upperquartile&quot;)
upperquartile &lt;- cpm(upperquartile, log=TRUE, normalized.lib.sizes = T)
head(upperquartile)</code></pre>
<pre><code>                18855_3_S 19160_3_S 18856_3_U 18856_1_U 18855_2_S 18856_2_S
ENSG00000223972 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000227232 -1.376624 -2.046788 -2.154778 -2.585632 -2.545595 -1.167335
ENSG00000278267 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000243485 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000284332 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000237613 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
                19160_3_U 18855_2_U 19160_2_S 18855_1_S 18856_1_S 19160_1_S
ENSG00000223972 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000227232 -3.129986 -2.527469 -1.317333 -1.415729 -3.129986 -2.313104
ENSG00000278267 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000243485 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000284332 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000237613 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
                19160_2_U 18855_1_U 18856_3_S 18856_2_U 18855_3_U
ENSG00000223972 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000227232 -2.601918 -1.247154 -2.084127 -1.609909 -3.129986
ENSG00000278267 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000243485 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000284332 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986
ENSG00000237613 -3.129986 -3.129986 -3.129986 -3.129986 -3.129986</code></pre>
<pre class="r"><code>strained &lt;- sampleinfo$treatment == &quot;Strain&quot;
unstrained &lt;- sampleinfo$treatment == &quot;Unstrain&quot;
ind_1 &lt;- sampleinfo$Individual == &quot;NA18855 &quot;
ind_2 &lt;- sampleinfo$Individual == &quot;NA18856 &quot;
ind_3 &lt;- sampleinfo$Individual == &quot;NA19160 &quot;


# Look at density plots for all individuals broken down by each treatment type
col = as.data.frame(pal[as.numeric(sampleinfo$Individual)])

plotDensities(upperquartile[,strained], col=col[strained, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-1">
Past versions of “upperquartile normalization-1.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization-1.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(upperquartile[,unstrained], col=col[unstrained, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-2">
Past versions of “upperquartile normalization-2.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization-2.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Look at density plots broken down by individual
col = as.data.frame(pal[as.numeric(sampleinfo$treatment)])

plotDensities(upperquartile[,ind_1], col=col[ind_1, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-3">
Past versions of “upperquartile normalization-3.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization-3.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(upperquartile[,ind_2], col=col[ind_2, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-4">
Past versions of “upperquartile normalization-4.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization-4.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(upperquartile[,ind_3], col=col[ind_3, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-5">
Past versions of “upperquartile normalization-5.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization-5.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="boxplots-of-upperquartile-across-samples" class="section level1">
<h1>Boxplots of upperquartile across samples</h1>
<pre class="r"><code>meltupperquartile &lt;- melt(upperquartile)
names(meltupperquartile) &lt;- c(&quot;gene&quot;, &quot;sampleID&quot;, &quot;upperquartile&quot;)
p &lt;- ggplot(meltupperquartile, aes(factor(sampleID), upperquartile)) 
p + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/boxplot%20upperquartile-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-upperquartile-1">
Past versions of “boxplot upperquartile-1.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-upperquartile-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/boxplot upperquartile-1.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filtering-for-lowly-expressed-genes" class="section level1">
<h1>Filtering for lowly expressed genes</h1>
<p>The criteria we are using is an average log2upperquartile &gt; 2.5 in at least 4 samples. After this filtering step, 10486 genes remain.</p>
<pre class="r"><code>cutoff &lt;- 2.5

keep &lt;- rowSums( upperquartile &gt; cutoff ) &gt;=4

counts_upperquartile &lt;- raw_counts[keep,]
filtered_upperquartile &lt;- upperquartile[keep,]
dim(filtered_upperquartile)</code></pre>
<pre><code>[1] 10486    17</code></pre>
</div>
<div id="boxplots-of-normalizedfiltered-counts-across-samples" class="section level1">
<h1>Boxplots of normalized+filtered counts across samples</h1>
<p>After normalization, the data do indeed look more normal than when unnormalized.</p>
<pre class="r"><code>melt_filt_upperquartile &lt;- melt(filtered_upperquartile)
names(melt_filt_upperquartile) &lt;- c(&quot;gene&quot;, &quot;sampleID&quot;, &quot;log2upperquartile&quot;)
p1 &lt;- ggplot(melt_filt_upperquartile, aes(factor(sampleID), log2upperquartile)) 
p1 + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/boxplot%20filt%20upperquartile-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-filt-upperquartile-1">
Past versions of “boxplot filt upperquartile-1.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-filt-upperquartile-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/boxplot filt upperquartile-1.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(filtered_upperquartile, legend = F)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/boxplot%20filt%20upperquartile-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-filt-upperquartile-2">
Past versions of “boxplot filt upperquartile-2.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-filt-upperquartile-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/0711682d4c989bade81cb5de5cdf2520fd5f9d33/docs/figure/normFilter_bulkRNA.Rmd/boxplot filt upperquartile-2.png" target="_blank">0711682</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="save-normalizedfiltered-count-matrix-and-filtered-count-matrix" class="section level1">
<h1>Save normalized/filtered count matrix and filtered count matrix</h1>
<pre class="r"><code>saveRDS(counts_upperquartile, &quot;data/filtered_counts.rds&quot;)
saveRDS(filtered_upperquartile, &quot;data/norm_filtered_counts.rds&quot;)</code></pre>
</div>
<div id="pca-plots-for-dataset-including-the-sample-filtered-out-by-qc" class="section level1">
<h1>PCA plots for dataset including the sample filtered out by QC</h1>
<div id="upperquartile-normalization-of-full-data" class="section level2">
<h2>upperquartile Normalization of full data</h2>
<pre class="r"><code>upperquartile_includeNA19160 &lt;- calcNormFactors(raw_counts_includeNA19160, method = &quot;upperquartile&quot;)
upperquartile_includeNA19160 &lt;- cpm(upperquartile_includeNA19160, log=TRUE, normalized.lib.sizes = T)
head(upperquartile_includeNA19160)</code></pre>
<pre><code>                18855_3_S 19160_3_S 18856_3_U 18856_1_U 18855_2_S 18856_2_S
ENSG00000223972 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000227232 -1.371277 -2.027812 -2.134933 -2.565487 -2.526602 -1.157511
ENSG00000278267 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000243485 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000284332 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000237613 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
                19160_3_U 18855_2_U 19160_2_S 18855_1_S 18856_1_S 19160_1_S
ENSG00000223972 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000227232 -3.101145 -2.508875 -1.294699 -1.401586 -3.101145 -2.296780
ENSG00000278267 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000243485 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000284332 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000237613 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
                19160_2_U 19160_1_U 18855_1_U 18856_3_S 18856_2_U 18855_3_U
ENSG00000223972 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000227232 -2.577970 -2.336134 -1.239235 -2.062052 -1.601659 -3.101145
ENSG00000278267 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000243485 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000284332 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145
ENSG00000237613 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145 -3.101145</code></pre>
<pre class="r"><code>strained &lt;- sampleinfo_includeNA19160$treatment == &quot;Strain&quot;
unstrained &lt;- sampleinfo_includeNA19160$treatment == &quot;Unstrain&quot;
ind_1 &lt;- sampleinfo_includeNA19160$Individual == &quot;NA18855 &quot;
ind_2 &lt;- sampleinfo_includeNA19160$Individual == &quot;NA18856 &quot;
ind_3 &lt;- sampleinfo_includeNA19160$Individual == &quot;NA19160 &quot;


# Look at density plots for all individuals broken down by each treatment type
col = as.data.frame(pal[as.numeric(sampleinfo_includeNA19160$Individual)])

plotDensities(upperquartile_includeNA19160[,strained], col=col[strained, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization%20full%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-full-data-1">
Past versions of “upperquartile normalization full data-1.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-full-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization full data-1.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(upperquartile_includeNA19160[,unstrained], col=col[unstrained, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization%20full%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-full-data-2">
Past versions of “upperquartile normalization full data-2.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-full-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization full data-2.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># Look at density plots broken down by individual
col = as.data.frame(pal[as.numeric(sampleinfo_includeNA19160$treatment)])

plotDensities(upperquartile_includeNA19160[,ind_1], col=col[ind_1, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization%20full%20data-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-full-data-3">
Past versions of “upperquartile normalization full data-3.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-full-data-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization full data-3.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(upperquartile_includeNA19160[,ind_2], col=col[ind_2, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization%20full%20data-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-full-data-4">
Past versions of “upperquartile normalization full data-4.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-full-data-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization full data-4.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(upperquartile_includeNA19160[,ind_3], col=col[ind_3, ], legend=&quot;topright&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/upperquartile%20normalization%20full%20data-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-upperquartile-normalization-full-data-5">
Past versions of “upperquartile normalization full data-5.png”
</button>
</p>
<div id="fig-no-spaces-upperquartile-normalization-full-data-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/upperquartile normalization full data-5.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="boxplots-of-upperquartile-across-samples-1" class="section level2">
<h2>Boxplots of upperquartile across samples</h2>
<pre class="r"><code>meltupperquartile_includeNA19160 &lt;- melt(upperquartile_includeNA19160)
names(meltupperquartile_includeNA19160) &lt;- c(&quot;gene&quot;, &quot;sampleID&quot;, &quot;upperquartile&quot;)
p &lt;- ggplot(meltupperquartile_includeNA19160, aes(factor(sampleID), upperquartile_includeNA19160)) 
p + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/boxplot%20upperquartile%20full%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-upperquartile-full-data-1">
Past versions of “boxplot upperquartile full data-1.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-upperquartile-full-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/boxplot upperquartile full data-1.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filtering-for-lowly-expressed-genes-avg-log2upperquartile-2.5-in-at-least-4-samples" class="section level2">
<h2>Filtering for lowly expressed genes (avg log2upperquartile &gt; 2.5 in at least 4 samples)</h2>
<pre class="r"><code>cutoff &lt;- 2.5

keep &lt;- rowSums( upperquartile_includeNA19160 &gt; cutoff ) &gt;=4

counts_upperquartile_includeNA19160 &lt;- raw_counts_includeNA19160[keep,]
filtered_upperquartile_includeNA19160 &lt;- upperquartile_includeNA19160[keep,]
dim(filtered_upperquartile_includeNA19160)</code></pre>
<pre><code>[1] 10517    18</code></pre>
</div>
<div id="boxplots-of-normalizedfiltered-counts-across-samples-1" class="section level2">
<h2>Boxplots of normalized+filtered counts across samples</h2>
<pre class="r"><code>melt_filt_upperquartile_includeNA19160 &lt;- melt(filtered_upperquartile_includeNA19160)
names(melt_filt_upperquartile_includeNA19160) &lt;- c(&quot;gene&quot;, &quot;sampleID&quot;, &quot;log2upperquartile&quot;)
p1 &lt;- ggplot(melt_filt_upperquartile_includeNA19160, aes(factor(sampleID), log2upperquartile)) 
p1 + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/boxplot%20filt%20upperquartile%20full%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-filt-upperquartile-full-data-1">
Past versions of “boxplot filt upperquartile full data-1.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-filt-upperquartile-full-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/boxplot filt upperquartile full data-1.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>plotDensities(filtered_upperquartile_includeNA19160, legend = F)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/boxplot%20filt%20upperquartile%20full%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-boxplot-filt-upperquartile-full-data-2">
Past versions of “boxplot filt upperquartile full data-2.png”
</button>
</p>
<div id="fig-no-spaces-boxplot-filt-upperquartile-full-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/boxplot filt upperquartile full data-2.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="plot-pca" class="section level2">
<h2>Plot PCA</h2>
<p>Reassuringly, the sample that was removed for low numbers of mapped reads still clusters as expected with its technical replicates.</p>
<pre class="r"><code>#Load PCA plotting Function
source(&quot;code/PCA_fn.R&quot;)

# Perform PCA
pca_genes &lt;- prcomp(t(filtered_upperquartile_includeNA19160), scale = T)
scores &lt;- pca_genes$x

variances &lt;- pca_genes$sdev^2
explained &lt;- variances / sum(variances)
plot(pca_genes, main = &quot;Variance per PC&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-1">
Past versions of “PCA full data-1.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-1.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#Make PCA plots with the factors colored by Individual

### PCA norm+filt Data
for (n in 1:5){
  col.v &lt;- pal[as.integer(sampleinfo_includeNA19160$Individual)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-2">
Past versions of “PCA full data-2.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-2.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-3">
Past versions of “PCA full data-3.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-3.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-4">
Past versions of “PCA full data-4.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-4.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-5">
Past versions of “PCA full data-5.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-5" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-5.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-6.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-6">
Past versions of “PCA full data-6.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-6" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-6.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#make PCA plots with symbols as treatment status and colors as individuals for figure
library(ggfortify)
autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = &quot;Individual&quot;, shape = &quot;treatment&quot;, size = 3) + 
     theme_cowplot() +
     theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-7.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-7">
Past versions of “PCA full data-7.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-7" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-7.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = &quot;Individual&quot;, shape = &quot;treatment&quot;) + 
     theme_cowplot()</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-8.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-8">
Past versions of “PCA full data-8.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-8" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-8.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = &quot;Individual&quot;, shape = &quot;treatment&quot;, size = 3, x = 3, y = 4) + 
     theme_cowplot() +
     theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-9">
Past versions of “PCA full data-9.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-9" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-9.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = &quot;Individual&quot;, shape = &quot;treatment&quot;, size = 3, x = 5, y = 6) + 
     theme_cowplot() +
     theme(legend.position = &quot;none&quot;)</code></pre>
<p><img src="figure/normFilter_bulkRNA.Rmd/PCA%20full%20data-10.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-PCA-full-data-10">
Past versions of “PCA full data-10.png”
</button>
</p>
<div id="fig-no-spaces-PCA-full-data-10" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/anthonyhung/invitroOA_pilot_repository/blob/e3fbd0fe2858abab87d6cc265d85f30638d8e531/docs/figure/normFilter_bulkRNA.Rmd/PCA full data-10.png" target="_blank">e3fbd0f</a>
</td>
<td>
Anthony Hung
</td>
<td>
2021-01-21
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] ggfortify_0.4.10   dplyr_1.0.2        cowplot_1.1.1      scales_1.1.1      
 [5] RColorBrewer_1.1-2 edgeR_3.26.8       limma_3.40.6       reshape_0.8.8     
 [9] ggplot2_3.3.2      gplots_3.0.4      

loaded via a namespace (and not attached):
 [1] gtools_3.8.2           tidyselect_1.1.0       locfit_1.5-9.4        
 [4] xfun_0.16              purrr_0.3.4            lattice_0.20-41       
 [7] colorspace_1.4-1       vctrs_0.3.4            generics_0.0.2        
[10] htmltools_0.5.0        yaml_2.2.1             rlang_0.4.7           
[13] later_1.1.0.1          pillar_1.4.6           glue_1.4.2            
[16] withr_2.2.0            lifecycle_0.2.0        plyr_1.8.6            
[19] stringr_1.4.0          munsell_0.5.0          gtable_0.3.0          
[22] workflowr_1.6.2        caTools_1.18.0         evaluate_0.14         
[25] labeling_0.3           knitr_1.29             httpuv_1.5.4          
[28] Rcpp_1.0.5             KernSmooth_2.23-17     promises_1.1.1        
[31] backports_1.1.9        gdata_2.18.0           farver_2.0.3          
[34] fs_1.5.0               gridExtra_2.3          digest_0.6.25         
[37] stringi_1.4.6          grid_3.6.3             rprojroot_1.3-2       
[40] tools_3.6.3            bitops_1.0-6           magrittr_1.5          
[43] tibble_3.0.3           tidyr_1.1.2            crayon_1.3.4          
[46] whisker_0.4            pkgconfig_2.0.3        ellipsis_0.3.1        
[49] rmarkdown_2.3          rstudioapi_0.11.0-9000 R6_2.4.1              
[52] git2r_0.27.1           compiler_3.6.3        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
