---
title: "Normalization and filtering of bulkRNAseq data"
author: "Anthony Hung"
date: "2021-01-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

This code takes the raw count data from bulk RNA sequencing and performs upper quartile normalziation on the data, filtering out lowly expressed genes. It then saves the resulting data and filtered count data (unnormalized) as R objects.

Here we also plot PCA plots to visualize the structure of the full data set (PCA plots for the data set excluding the sample that failed QC are located in the [Modeling and removing unwanted technical variation](removeUnwantedVariation_bulkRNA.html) page).

# Load data and libraries

Here we are loading both the full data and the data without the one sample that failed QC.

```{r load libraries and data}
library("gplots")
library("ggplot2")
library("reshape")
library("edgeR")
library("RColorBrewer")
library("scales")
library("cowplot")
theme_set(theme_cowplot())
library("dplyr")

# Load colors 
colors <- colorRampPalette(c(brewer.pal(9, "Blues")[1],brewer.pal(9, "Blues")[9]))(100)
pal <- c(brewer.pal(9, "Set1"), brewer.pal(8, "Set2"), brewer.pal(12, "Set3"))


# load in relabeled counts
raw_counts <- readRDS("data/raw_counts_relabeled.rds")
raw_counts_includeNA19160 <- readRDS("data/raw_counts_relabeled_includeNA19160.rds")

# Create DGEList object to allow for easier application of different normalization methods
raw_counts <- DGEList(raw_counts, group = colnames(raw_counts))
raw_counts_includeNA19160 <- DGEList(raw_counts_includeNA19160, group = colnames(raw_counts_includeNA19160))

# load in reordered sample information
sampleinfo <- readRDS("data/Sample.info.RNAseq.reordered.rds")
sampleinfo_includeNA19160 <- readRDS("data/Sample.info.RNAseq.reordered_includeNA19160.rds")
```

# Upperquartile Normalization

Apply upperquartile normalization to the data.

```{r upperquartile normalization}
upperquartile <- calcNormFactors(raw_counts, method = "upperquartile")
upperquartile <- cpm(upperquartile, log=TRUE, normalized.lib.sizes = T)
head(upperquartile)

strained <- sampleinfo$treatment == "Strain"
unstrained <- sampleinfo$treatment == "Unstrain"
ind_1 <- sampleinfo$Individual == "NA18855 "
ind_2 <- sampleinfo$Individual == "NA18856 "
ind_3 <- sampleinfo$Individual == "NA19160 "


# Look at density plots for all individuals broken down by each treatment type
col = as.data.frame(pal[as.numeric(sampleinfo$Individual)])

plotDensities(upperquartile[,strained], col=col[strained, ], legend="topright")
plotDensities(upperquartile[,unstrained], col=col[unstrained, ], legend="topright")

# Look at density plots broken down by individual
col = as.data.frame(pal[as.numeric(sampleinfo$treatment)])

plotDensities(upperquartile[,ind_1], col=col[ind_1, ], legend="topright")
plotDensities(upperquartile[,ind_2], col=col[ind_2, ], legend="topright")
plotDensities(upperquartile[,ind_3], col=col[ind_3, ], legend="topright")
```

# Boxplots of upperquartile across samples

```{r boxplot upperquartile}
meltupperquartile <- melt(upperquartile)
names(meltupperquartile) <- c("gene", "sampleID", "upperquartile")
p <- ggplot(meltupperquartile, aes(factor(sampleID), upperquartile)) 
p + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))
```

# Filtering for lowly expressed genes

The criteria we are using is an average log2upperquartile > 2.5 in at least 4 samples. After this filtering step, 10486 genes remain.

```{r filter upperquartile}
cutoff <- 2.5

keep <- rowSums( upperquartile > cutoff ) >=4

counts_upperquartile <- raw_counts[keep,]
filtered_upperquartile <- upperquartile[keep,]
dim(filtered_upperquartile)
```

# Boxplots of normalized+filtered counts across samples

After normalization, the data do indeed look more normal than when unnormalized.

```{r boxplot filt upperquartile}
melt_filt_upperquartile <- melt(filtered_upperquartile)
names(melt_filt_upperquartile) <- c("gene", "sampleID", "log2upperquartile")
p1 <- ggplot(melt_filt_upperquartile, aes(factor(sampleID), log2upperquartile)) 
p1 + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))

plotDensities(filtered_upperquartile, legend = F)
```

# Save normalized/filtered count matrix and filtered count matrix

```{r save}
saveRDS(counts_upperquartile, "data/filtered_counts.rds")
saveRDS(filtered_upperquartile, "data/norm_filtered_counts.rds")
```

# PCA plots for dataset including the sample filtered out by QC

## upperquartile Normalization of full data

```{r upperquartile normalization full data}
upperquartile_includeNA19160 <- calcNormFactors(raw_counts_includeNA19160, method = "upperquartile")
upperquartile_includeNA19160 <- cpm(upperquartile_includeNA19160, log=TRUE, normalized.lib.sizes = T)
head(upperquartile_includeNA19160)

strained <- sampleinfo_includeNA19160$treatment == "Strain"
unstrained <- sampleinfo_includeNA19160$treatment == "Unstrain"
ind_1 <- sampleinfo_includeNA19160$Individual == "NA18855 "
ind_2 <- sampleinfo_includeNA19160$Individual == "NA18856 "
ind_3 <- sampleinfo_includeNA19160$Individual == "NA19160 "


# Look at density plots for all individuals broken down by each treatment type
col = as.data.frame(pal[as.numeric(sampleinfo_includeNA19160$Individual)])

plotDensities(upperquartile_includeNA19160[,strained], col=col[strained, ], legend="topright")
plotDensities(upperquartile_includeNA19160[,unstrained], col=col[unstrained, ], legend="topright")

# Look at density plots broken down by individual
col = as.data.frame(pal[as.numeric(sampleinfo_includeNA19160$treatment)])

plotDensities(upperquartile_includeNA19160[,ind_1], col=col[ind_1, ], legend="topright")
plotDensities(upperquartile_includeNA19160[,ind_2], col=col[ind_2, ], legend="topright")
plotDensities(upperquartile_includeNA19160[,ind_3], col=col[ind_3, ], legend="topright")
```

## Boxplots of upperquartile across samples

```{r boxplot upperquartile full data}
meltupperquartile_includeNA19160 <- melt(upperquartile_includeNA19160)
names(meltupperquartile_includeNA19160) <- c("gene", "sampleID", "upperquartile")
p <- ggplot(meltupperquartile_includeNA19160, aes(factor(sampleID), upperquartile_includeNA19160)) 
p + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))
```

## Filtering for lowly expressed genes (avg log2upperquartile > 2.5 in at least 4 samples)

```{r filter upperquartile full data}
cutoff <- 2.5

keep <- rowSums( upperquartile_includeNA19160 > cutoff ) >=4

counts_upperquartile_includeNA19160 <- raw_counts_includeNA19160[keep,]
filtered_upperquartile_includeNA19160 <- upperquartile_includeNA19160[keep,]
dim(filtered_upperquartile_includeNA19160)
```

## Boxplots of normalized+filtered counts across samples

```{r boxplot filt upperquartile full data}
melt_filt_upperquartile_includeNA19160 <- melt(filtered_upperquartile_includeNA19160)
names(melt_filt_upperquartile_includeNA19160) <- c("gene", "sampleID", "log2upperquartile")
p1 <- ggplot(melt_filt_upperquartile_includeNA19160, aes(factor(sampleID), log2upperquartile)) 
p1 + geom_boxplot() + theme(axis.text.x = element_text(angle = 90))

plotDensities(filtered_upperquartile_includeNA19160, legend = F)
```

## Plot PCA

Reassuringly, the sample that was removed for low numbers of mapped reads still clusters as expected with its technical replicates.

```{r PCA full data}
#Load PCA plotting Function
source("code/PCA_fn.R")

# Perform PCA
pca_genes <- prcomp(t(filtered_upperquartile_includeNA19160), scale = T)
scores <- pca_genes$x

variances <- pca_genes$sdev^2
explained <- variances / sum(variances)
plot(pca_genes, main = "Variance per PC")


#Make PCA plots with the factors colored by Individual

### PCA norm+filt Data
for (n in 1:5){
  col.v <- pal[as.integer(sampleinfo_includeNA19160$Individual)]
  plot_scores(pca_genes, scores, n, n+1, col.v)
}

#make PCA plots with symbols as treatment status and colors as individuals for figure
library(ggfortify)
autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = "Individual", shape = "treatment", size = 3) + 
     theme_cowplot() +
     theme(legend.position = "none")
autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = "Individual", shape = "treatment") + 
     theme_cowplot()
autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = "Individual", shape = "treatment", size = 3, x = 3, y = 4) + 
     theme_cowplot() +
     theme(legend.position = "none")
autoplot(pca_genes, data = sampleinfo_includeNA19160, colour = "Individual", shape = "treatment", size = 3, x = 5, y = 6) + 
     theme_cowplot() +
     theme(legend.position = "none")
```
