---
title: "Pre-processing of raw bulk RNA sequencing files"
author: "Anthony Hung"
date: "2021-01-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

This page will walk through the steps to go from the raw bulk RNA sequencing fastq files to a count matrix. This involves running a Snakemake pipeline located in the code directory and some code in R. The end product is a count matrix corresponding to genes x samples containing information from all 18 bulk samples from the study. We will also load in sample meta information from a .csv file in this repository.

# Run the Snakemake Pipeline after downloading necessary files and creating the conda environment (see README.txt)

1. Move the fastq files from all samples (these are output files from bulk RNA sequencing runs) into the folder
code/bulkRNA_preprocessing/fastq/. Undetermined data files are not required.

2. Modify the code/bulkRNA_preprocessing/Pipeline/cluster_solo.json file to correspond to the computing cluster you are working with.

3. Create a STAR genome index for hg38 and point to its location in the config_bulk_hg38.yaml file.

4. Download a GTF file for hg38 and point to its location in the config_bulk_hg38.yaml file.

5. Install the conda working by running
"conda env create --file environment.yaml"

6. Run "source activate bulkRNAseq" to activate the bulkRNAseq conda environment.

7. Run the file "submit.sh".

# Load files into R

After running the Snakemake pipeline, you will have output files (located in code/bulkRNA_preprocessing/out) corresponding to the count matrix and quality control of the raw fastq files. In this repository is also included sample information in a csv. We will now load the count matrix and sample information into R and save them as R objects for subsequent analysis.
# Load in bulk RNA sequencing data from pilot experiment and visualize correlations between samples 

```{r load libraries and data}
library("gplots")
library("ggplot2")
library("RColorBrewer")
library("scales")
library("edgeR")

#These sample names correspond to the individual_replicate_treatmentCondition.
samplenames <- c("18855_3_S","19160_3_S", "18856_3_U",
                 "18856_1_U","18855_2_S", "18856_2_S",
                 "19160_3_U","18855_2_U", "19160_2_S",
                 "18855_1_S","18856_1_S", "19160_1_S",
                 "19160_2_U","19160_1_U", "18855_1_U",
                 "18856_3_S","18856_2_U", "18855_3_U"
                 )

# Load colors 
colors <- colorRampPalette(c(brewer.pal(9, "Blues")[1],brewer.pal(9, "Blues")[9]))(100)
pal <- c(brewer.pal(9, "Set1"), brewer.pal(8, "Set2"), brewer.pal(12, "Set3"))

# load in counts
raw_counts <- read.table("code/bulkRNA_preprocessing/out/counts/counts.txt", header = T)

# assign row.names based on the Geneid column
row.names(raw_counts) <- raw_counts$Geneid

# remove extra data columns from count matrix
raw_counts <- raw_counts[, -c(1:6)] 

# assign col.names 
names(raw_counts) <- samplenames

#Sample information
sampleinfo <- read.csv("data/Sample.info.RNAseq.csv")
head(sampleinfo)
#Re-order sample information to match the columns of count matrix
sampleinfo <- sampleinfo[match(samplenames, sampleinfo$Sample_ID),]
```


# Based on the QC metrics, one of our samples did not pass with high quality and is therefore removed from subsequent analyses. We will, however, keep a copy of the full data to demonstrate that the sample does cluster as expected with its corresponding technical replicates (NA19160 Unstrain, first Replicate).

```{r remove sample}
# save relabeled raw count data matrix
saveRDS(raw_counts, "data/raw_counts_relabeled_includeNA19160.rds")
# save reorganized sample information
saveRDS(sampleinfo, "data/Sample.info.RNAseq.reordered_includeNA19160.rds")

# remove 02 sample that does not pass QC. More information in manuscript.
raw_counts <- raw_counts[, -14] 
#remove 02 sample that does not pass QC
sampleinfo <- sampleinfo[-14,]
```


# Save the raw count and sample information data

```{r Save R objects}
# save relabeled raw count data matrix
saveRDS(raw_counts, "data/raw_counts_relabeled.rds")
# save reorganized sample information
saveRDS(sampleinfo, "data/Sample.info.RNAseq.reordered.rds")
```


