---
title: "Variance partitioning the bulk RNAseq data"
author: "Anthony Hung"
date: "2021-01-19"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: console
---

# Introduction

This code uses the variancePartition package to quantify the percent of variance in the data explained by some biological/technical factors.

# Load packages and normalized data

```{r load}
library(variancePartition)
library(RUVSeq)
library(edgeR)
library(dplyr)

#raw(filtered) counts
counts_upperquartile <- readRDS("data/filtered_counts.rds")$counts
#normalized and filtered counts
filtered_upperquartile <- readRDS("data/norm_filtered_counts.rds")
#metadata
sampleinfo <- readRDS("data/Sample.info.RNAseq.reordered.rds")
```

# Fitting unwanted variation

Before fitting the linear mixed model to quantify contributions of different variables to variation in the dataset, we are also interested in fitting one factor of unwanted variation to include in the model using RUVg. For the RUVg control genes, we use the top 100 (~1% of the genes considered) LEAST variable genes as control genes.

```{r identify least variable genes}
#input data consists of raw filtered data (filtered for lowly expressed genes)
#compute CV (stdev/mean) and rank from least to most; pick 100 least variable
cv <- apply(counts_upperquartile, 1, function(x) sd(x)/mean(x))
least_var_genes <- names(head(sort(cv), 100))
```

Apply RUVseq with the set of least variable genes to determine a single factor of unwanted variation.

```{r RUVSeq least variable genes}
#load data into expressionset
set <- newSeqExpressionSet(as.matrix(counts_upperquartile),phenoData = data.frame(sampleinfo, row.names=colnames(counts_upperquartile)))
set
#normalization
set <- betweenLaneNormalization(x = set, which = "upper")
#run RUVg
set1 <- RUVg(set, least_var_genes, k=1)
sample_info <- pData(set1)
sample_info$Replicate <- as.factor(sample_info$Replicate)
sample_info$LibraryPrepBatch <- as.factor(sample_info$LibraryPrepBatch)
```

# Specify variables to consider in LMM

Here, we specify two separate models (one including the single factor of unwanted variation computed above and one without this factor).

```{r variables_to_consider}
# Specify variables to consider
form <- ~ (1|Individual) + (1|treatment) + (1|Replicate) + (1|LibraryPrepBatch) + W_1

# No ruv
form_no_ruv <- ~ (1|Individual) + (1|treatment) + (1|Replicate) + (1|LibraryPrepBatch)
```

# Partition the variance and plot the result

```{r function_1}
# Fit model and extract results
# 1) fit linear mixed model on gene expression
# If categorical variables are specified,
# a linear mixed model is used
# If all variables are modeled as fixed effects,
# a linear model is used
# each entry in results is a regression model fit on a single gene
# 2) extract variance fractions from each model fit
# for each gene, returns fraction of variation attributable
# to each variable
# Interpretation: the variance explained by each variables
# after correcting for all other variables
varPart_1_int <- fitExtractVarPartModel( filtered_upperquartile, form, sample_info )
# sort variables (i.e. columns) by median fraction
# of variance explained
vp <- sortCols(varPart_1_int)
# violin plot of contribution of each variable to total variance
plotVarPart( vp )

#noRuv
VarPart_noRUV <- fitExtractVarPartModel( filtered_upperquartile, form_no_ruv, sample_info )
vp_noRUV <- sortCols(VarPart_noRUV)
plotVarPart( vp_noRUV )
```

